snap/hooks/install

```diff
@@ -135,11 +134,43 @@ if [ ! -f "$SNAP_DATA/config/security-api-gateway/kong.conf" ]; then
 
     # also replace the default nginx user/group to be root
     sed -i "s@#nginx_user = nobody nobody@nginx_user = root root@" "$SNAP_DATA/config/security-api-gateway/kong.conf"
+fi
 
-    # finally set the make database to be cassandra
-    sed -i "s@#database = postgres@database = cassandra@" "$SNAP_DATA/config/security-api-gateway/kong.conf"
+# setup postgres db config file with env vars replaced
+if [ ! -f "$SNAP_DATA/etc/postgresql/10/main/postgresql.conf" ]; then
+    mkdir -p "$SNAP_DATA/etc/postgresql/10/main"
+    cp "$SNAP/etc/postgresql/10/main/postgresql.conf" "$SNAP_DATA/etc/postgresql/10/main/postgresql.conf"
+    # do replacement of the $SNAP, $SNAP_DATA, $SNAP_COMMON environment variables in the config files
+    sed -i -e "s@\$SNAP_COMMON@$SNAP_COMMON@g" \
+        -e "s@\$SNAP_DATA@$SNAP_DATA_CURRENT@g" \
+        -e "s@\$SNAP@$SNAP_CURRENT@g" \
+        "$SNAP_DATA/etc/postgresql/10/main/postgresql.conf"
 fi
 
+# ensure the postgres data directory exists and is owned by snap_daemon
+mkdir -p "$SNAP_DATA/postgresql" 
+chown -R snap_daemon:snap_daemon "$SNAP_DATA/postgresql" 
+
+# setup the postgres data directory
+gosu snap_daemon "$SNAP/usr/lib/postgresql/10/bin/initdb" -D "$SNAP_DATA/postgresql/10/main"
+
+# ensure the sockets dir exists and is properly owned
+mkdir -p "$SNAP_COMMON/sockets"
+chown -R snap_daemon:snap_daemon "$SNAP_COMMON/sockets" 
+
+# start postgres up and wait a bit for it so we can create the database and user
+# for kong
+snapctl start "$SNAP_NAME.postgres"
+sleep 2
```

Comment:
Is there a cleaner approach to this? Problems related to sleep commands are notoriously hard to track down if/when they go wrong...

Reply:
:thinking: well the problem is that we need to run the postgres commands below to bootstrap postgres for kong, which talk to the postgres daemon, so I guess we could loop trying to run the commands until they succeed instead with the sleep there instead. Let me try that out

Reply:
yes looping works fine, I went with this approach